hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java

```diff
@@ -324,6 +325,11 @@ private LoadBalancer getBalancer() {
     return master.getLoadBalancer();
   }
 
+  private FavoredNodesPromoter getFavoredNodePromoter() {
+    return (FavoredNodesPromoter) ((RSGroupBasedLoadBalancer) master.getLoadBalancer())
```

Comment:
No need to cast to RSGroupBasedLoadBalancer?

Reply:
The return value for the MasterServices interface is still LoadBalancer so...

Reply:
For 3.0, I thought we can change this in the future?